### PR TITLE
Use opentxs::identifier::Nym class for nym identifiers

### PR DESCRIPTION
**Pull request by Wallydog66**
Signed-off-by: Kevin Waldren <wallydog.66@gmail.com>

### CODE REVIEW

File: src/client/OT_API.cpp [hunk 1]
```diff
@@ -865,12 +866,12 @@ bool OT_API::Wallet_CanRemoveServer(const Identifier& NOTARY_ID) const
 
     // Loop through all the Nyms. (One might be registered on that server.)
     //
-    std::set<OTIdentifier> nymIDs = api_.Wallet().LocalNyms();
+    std::set<OTNymID> nymIDs = api_.Wallet().LocalNyms();
     for (auto& nymID : nymIDs) {
-        if (IsNym_RegisteredAtServer(nymID, NOTARY_ID)) {
+        if (IsNym_RegisteredAtServer(identifier::Nym::Factory(nymID->str()), NOTARY_ID)) {
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 2]
```diff
@@ -865,12 +866,12 @@ bool OT_API::Wallet_CanRemoveServer(const Identifier& NOTARY_ID) const
...
             LogOutput(OT_METHOD)(__FUNCTION__)(
                 ": Unable to remove server contract ")(NOTARY_ID)(
-                " from wallet, because Nym ")(nymID)(
+                " from wallet, because Nym ")(nymID->str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 3]
```diff
@@ -940,7 +941,7 @@ bool OT_API::Wallet_CanRemoveNym(const Identifier& NYM_ID) const
     const auto accounts = api_.Storage().AccountsByOwner(NYM_ID);
 
     if (0 < accounts.size()) {
-        LogOutput(OT_METHOD)(__FUNCTION__)(": Unable to remove nym ")(NYM_ID)(
+        LogOutput(OT_METHOD)(__FUNCTION__)(": Unable to remove nym ")(NYM_ID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 4]
```diff
@@ -1991,13 +1992,13 @@ bool OT_API::VerifyAndRetrieveXMLContents(
...
     const Identifier& NOTARY_ID,
-    const Identifier& NYM_ID,
+    const identifier::Nym& NYM_ID,
     const Identifier& ACCOUNT_ID) const
 {
     auto context = api_.Wallet().ServerContext(NYM_ID, NOTARY_ID);
 
     if (false == bool(context)) {
-        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID)(
+        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 5]
```diff
@@ -2735,7 +2736,7 @@ bool OT_API::SmartContract_ConfirmParty(
     //  pMessage->m_strNotaryID        = strNotaryID;
     pMessage->m_ascPayload->SetString(strInstrument);
 
-    auto pNym = api_.Wallet().Nym(nymfile.It().ID());
+    auto pNym = api_.Wallet().Nym(identifier::Nym::Factory());
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: bug

File: src/client/OT_API.cpp [hunk 6]
```diff
@@ -4176,13 +4177,13 @@ bool OT_API::ConfirmPaymentPlan(
...
     const Identifier& NOTARY_ID,
-    const Identifier& NYM_ID) const
+    const identifier::Nym& NYM_ID) const
 {
     rLock lock(lock_callback_({NYM_ID.str(), NOTARY_ID.str()}));
     auto context = api_.Wallet().ServerContext(NYM_ID, NOTARY_ID);
 
     if (false == bool(context)) {
-        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID)(
+        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 7]
```diff
@@ -4242,14 +4243,14 @@ std::unique_ptr<Ledger> OT_API::LoadNymboxNoVerify(
...
-    const Identifier& NYM_ID,
+    const identifier::Nym& NYM_ID,
     const Identifier& ACCOUNT_ID) const
 {
     rLock lock(lock_callback_({NYM_ID.str(), NOTARY_ID.str()}));
     auto context = api_.Wallet().ServerContext(NYM_ID, NOTARY_ID);
 
     if (false == bool(context)) {
-        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID)(
+        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 8]
```diff
@@ -4320,14 +4321,14 @@ std::unique_ptr<Ledger> OT_API::LoadInboxNoVerify(
...
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 9]
```diff
@@ -4399,13 +4400,13 @@ std::unique_ptr<Ledger> OT_API::LoadOutboxNoVerify(
...
     const Identifier& NOTARY_ID,
-    const Identifier& NYM_ID) const
+    const identifier::Nym& NYM_ID) const
 {
     rLock lock(lock_callback_({NYM_ID.str(), NOTARY_ID.str()}));
     auto context = api_.Wallet().ServerContext(NYM_ID, NOTARY_ID);
 
     if (false == bool(context)) {
-        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID)(
+        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 10]
```diff
@@ -4465,14 +4466,14 @@ std::unique_ptr<Ledger> OT_API::LoadPaymentInboxNoVerify(
...
-    const Identifier& NYM_ID,
+    const identifier::Nym& NYM_ID,
     const Identifier& ACCOUNT_ID) const
 {
     rLock lock(lock_callback_({NYM_ID.str(), NOTARY_ID.str()}));
     auto context = api_.Wallet().ServerContext(NYM_ID, NOTARY_ID);
 
     if (false == bool(context)) {
-        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID)(
+        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 11]
```diff
@@ -4539,13 +4540,13 @@ std::unique_ptr<Ledger> OT_API::LoadRecordBoxNoVerify(
...
     const Identifier& NOTARY_ID,
-    const Identifier& NYM_ID) const
+    const identifier::Nym& NYM_ID) const
 {
     rLock lock(lock_callback_({NYM_ID.str(), NOTARY_ID.str()}));
     auto context = api_.Wallet().ServerContext(NYM_ID, NOTARY_ID);
 
     if (false == bool(context)) {
-        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID)(
+        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 12]
```diff
@@ -4617,7 +4618,7 @@ bool OT_API::ClearExpired(
...
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 13]
```diff
@@ -5712,9 +5713,9 @@ bool OT_API::RecordPayment(
                             // case, he's the "recipient." (So we need to figure
                             // out which, and set the account accordingly.)
                             //
-                            if (NYM_ID == pPlan->GetRecipientNymID())
+                            if (NYM_ID.str() == pPlan->GetRecipientNymID().str())
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 14]
```diff
@@ -5712,9 +5713,9 @@ bool OT_API::RecordPayment(
...
                                 theSenderAcctID = pPlan->GetRecipientAcctID();
-                            else if (NYM_ID == pPlan->GetSenderNymID())
+                            else if (NYM_ID.str() == pPlan->GetSenderNymID().str())
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 15]
```diff
@@ -6074,7 +6075,7 @@ bool OT_API::ClearRecord(
     auto context = api_.Wallet().ServerContext(NYM_ID, NOTARY_ID);
 
     if (false == bool(context)) {
-        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID)(
+        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 16]
```diff
@@ -6277,7 +6278,7 @@ void OT_API::FlushSentMessages(
     const auto strNotaryID = String::Factory(NOTARY_ID),
                strNymID = String::Factory(NYM_ID);
 
-    if ((THE_NYMBOX.GetNymID() != NYM_ID) ||
+    if ((THE_NYMBOX.GetNymID().str() != NYM_ID.str()) ||
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 17]
```diff
@@ -6345,13 +6346,13 @@ void OT_API::FlushSentMessages(
...
     const Identifier& NOTARY_ID,
-    const Identifier& NYM_ID,
+    const identifier::Nym& NYM_ID,
     const RequestNumber& lRequestNumber) const
 {
     auto context = api_.Wallet().ServerContext(NYM_ID, NOTARY_ID);
 
     if (false == bool(context)) {
-        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID)(
+        LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(NYM_ID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 18]
```diff
@@ -7900,7 +7901,7 @@ bool OT_API::DiscardCheque(
     } else if (
         (theCheque->GetNotaryID() == NOTARY_ID) &&
         (theCheque->GetInstrumentDefinitionID() == contractID) &&
-        (theCheque->GetSenderNymID() == NYM_ID) &&
+        (theCheque->GetSenderNymID().str() == NYM_ID.str()) &&
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 19]
```diff
@@ -7990,11 +7991,11 @@ CommandResult OT_API::depositCheque(
     if (theCheque.HasRemitter()) {
         bCancellingCheque =
             ((theCheque.GetRemitterAcctID() == accountID) &&
-             (theCheque.GetRemitterNymID() == nymID));
+             (theCheque.GetRemitterNymID().str() == nymID.str()));
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 20]
```diff
@@ -7990,11 +7991,11 @@ CommandResult OT_API::depositCheque(
...
     } else {
         bCancellingCheque =
             ((theCheque.GetSenderAcctID() == accountID) &&
-             (theCheque.GetSenderNymID() == nymID));
+             (theCheque.GetSenderNymID().str() == nymID.str()));
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 21]
```diff
@@ -8222,7 +8223,7 @@ CommandResult OT_API::depositPaymentPlan(
         return output;
     }
 
-    const bool bCancelling = (plan->GetRecipientNymID() == nymID);
+    const bool bCancelling = (plan->GetRecipientNymID().str() == nymID.str());
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 22]
```diff
@@ -9523,7 +9524,7 @@ CommandResult OT_API::processNymbox(ServerContext& context) const
     auto nymbox{api_.Factory().Ledger(nymID, nymID, serverID)};
 
     if (false == nymbox->LoadNymbox()) {
-        LogOutput(OT_METHOD)(__FUNCTION__)(": Failed loading Nymbox: ")(nymID)(
+        LogOutput(OT_METHOD)(__FUNCTION__)(": Failed loading Nymbox: ")(nymID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 23]
```diff
@@ -9532,14 +9533,14 @@ CommandResult OT_API::processNymbox(ServerContext& context) const
 
     if (false == nymbox->VerifyAccount(nym)) {
         LogOutput(OT_METHOD)(__FUNCTION__)(": Failed verifying Nymbox: ")(
-            nymID)(".")
+            nymID.str())(".")
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 24]
```diff
@@ -9532,14 +9533,14 @@ CommandResult OT_API::processNymbox(ServerContext& context) const
...
             .Flush();
 
         return output;
     }
 
     if (1 > nymbox->GetTransactionCount()) {
-        LogDetail(OT_METHOD)(__FUNCTION__)(": Nymbox (")(nymID)(
+        LogDetail(OT_METHOD)(__FUNCTION__)(": Nymbox (")(nymID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 25]
```diff
@@ -10039,7 +10040,7 @@ CommandResult OT_API::registerContract(
 
     switch (TYPE) {
         case (ContractType::NYM): {
-            const auto contract = api_.Wallet().Nym(CONTRACT);
+            const auto contract = api_.Wallet().Nym(identifier::Nym::Factory());
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: bug

File: src/client/OT_API.cpp [hunk 26]
```diff
@@ -10378,7 +10379,7 @@ CommandResult OT_API::sendNymInstrument(
     // We only actually SEND if sender and recipient are different
     // nyms. (If they're the same Nym, we instead only save a copy
     // in the outbox.)
-    if (nymID != recipientNymID) {
+    if (nymID.str() != recipientNymID.str()) {
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 27]
```diff
@@ -10512,7 +10513,7 @@ OT_API::ProcessInbox OT_API::Ledger_CreateResponse(
         auto context = api_.Wallet().ServerContext(theNymID, theNotaryID);
 
         if (false == bool(context)) {
-            LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(theNymID)(
+            LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(theNymID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OT_API.cpp [hunk 28]
```diff
@@ -11213,7 +11214,7 @@ CommandResult OT_API::initiatePeerRequest(
     }
 
     std::unique_ptr<Message> request{nullptr};
-    output = sendNymObject(context, request, recipient, *object, requestNum);
+    output = sendNymObject(context, request, identifier::Nym::Factory(), *object, requestNum);
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: bug

File: src/client/OT_API.cpp [hunk 29]
```diff
@@ -11254,7 +11255,7 @@ CommandResult OT_API::initiatePeerReply(
         return output;
     }
 
-    auto recipientNym = api_.Wallet().Nym(recipient);
+    auto recipientNym = api_.Wallet().Nym(identifier::Nym::Factory());
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: bug

File: src/client/OT_API.cpp [hunk 30]
```diff
@@ -11292,7 +11293,7 @@ CommandResult OT_API::initiatePeerReply(
 
     std::unique_ptr<Message> requestMessage{nullptr};
     output =
-        sendNymObject(context, requestMessage, recipient, *object, requestNum);
+        sendNymObject(context, requestMessage, identifier::Nym::Factory(), *object, requestNum);
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: bug

File: src/client/OT_API.cpp [hunk 31]
```diff
@@ -11965,7 +11966,7 @@ OTTransaction* OT_API::get_or_create_process_inbox(
             context.NextTransactionNumber(MessageType::processInbox);
 
         if (false == number->Valid()) {
-            LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(nymID)(
+            LogOutput(OT_METHOD)(__FUNCTION__)(": Nym ")(nymID.str())(
```
Comment (justusranvier) [inline src/client/OT_API.cpp]: unnecessary

File: src/client/OTAPI_Exec.cpp [hunk 32]
```diff
@@ -860,7 +861,7 @@ bool OTAPI_Exec::SetContactData(
             .Flush();
         return false;
     }
-    opentxs::OTIdentifier nymID = Identifier::Factory(NYM_ID);
+    opentxs::OTNymID nymID = identifier::Nym::Factory(NYM_ID);
```
Comment (justusranvier) [inline src/client/OTAPI_Exec.cpp]:
```suggestion
    auto nymID = identifier::Nym::Factory(NYM_ID);
```

File: src/server/UserCommandProcessor.cpp [hunk 33]
```diff
@@ -310,11 +311,11 @@ void UserCommandProcessor::check_acknowledgements(ReplyMessage& reply) const
 }
 
 bool UserCommandProcessor::check_client_isnt_server(
-    const Identifier& nymID,
+    const identifier::Nym& nymID,
     const Nym& serverNym)
 {
     const auto& serverNymID = serverNym.ID();
-    const bool bNymIsServerNym = serverNymID == nymID;
+    const bool bNymIsServerNym = serverNymID.str() == nymID.str();
```
[the remark this answers is not in the document]
Comment (Wallydog66) [reply src/server/UserCommandProcessor.cpp]: Have error following requested fix, request suggestion to rectify.

**Pull request closed without merging**